### PR TITLE
52 webpack and folder set up

### PR DESCRIPTION
**Pull request by qzyan**
[trello ticket](https://trello.com/c/hFjLbQv5/52-webpack-and-folder-set-up-s)  
- webpack set up 
- .gitignore set up  
- client/src/components set up

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+const path = require('path');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+
+module.exports = {
+  mode: "development",
+  entry: path.join(__dirname, "client/src", "index.js"),
+  output: {
+    path:path.resolve(__dirname, "client/dist"),
+  },
+  module: {
```
Comment (hepnerthomas) [inline webpack.config.js:10]: investigate if neceesary to add `devServer` to the module if you want webpack to serve the files locally.
https://webpack.js.org/configuration/dev-server/

Comment (hepnerthomas) [reply webpack.config.js:10]: https://webpack.js.org/configuration/dev-server/

**Review by hepnerthomas — CHANGES_REQUESTED** [review]
Great work, just two minor changes requested!

**Requested Changes**
- add example.env file w/ comments about how to create an .env file
- add `dotenv` to the dependencies in package.json

**Comment by allanviguilla** [comment]
@qzyan also just wanted to ask to include eslint as a dependency in package.json

File: exmple.env [hunk 2]
```diff
@@ -0,0 +1 @@
+API_TOKEN
```
Comment (hepnerthomas) [inline exmple.env]: @qzyan Can you change the filename to `example.env` really quick? You have a small typo.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    "reatail"
+  ],
+  "author": "qingzhou yan",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/qzyan/web-application-X/issues"
+  },
+  "homepage": "https://github.com/qzyan/web-application-X#readme",
+  "dependencies": {
+    "dotenv": "^16.0.1",
```
Comment (hepnerthomas) [inline package.json:25]: @qzyan nicely done!

**Review by hepnerthomas — APPROVED** [review]
All good - thank you for adding in the example.env file!

**Pull request merged**